examples/framework.rs

```diff
@@ -30,10 +30,13 @@ pub enum ShaderStage {
 }
 
 pub trait Example: 'static + Sized {
-    fn needed_features() -> wgpu::Features {
+    fn requested_features() -> wgpu::Features {
```

Comment:
maybe have it the other way, and call `optional_features`? This would let examples avoiding to repeat themselves if they need to implement both functions